src/shared_modules/content_manager/README.md

```diff
@@ -0,0 +1,341 @@
+  + `contentFileName`: Used by some downloaders to know where to store the downloaded content.
+  + `databasePath`: Path from where the RocksDB database should be read. The database stores the last offset fetched (when using the `cti-api` content source).
+
+### Use case: Download offsets from CTI API
+
+One of the most important capabilities of the Content Manager is to download content deltas (called _offsets_) from a _Cyber Threat Intelligence_ (CTI) API. The module will try to download all available offsets, starting from the last offset fetched from the previous execution (stored in the RocksDB database), until it reaches the last offset available in the API.
+
+All the downloaded content will be stored in the filesystem, making it available to other modules that may want to consume it.
+
+```json
+{
+    "topicName": "CTI API offset fetching",
+    "interval": 10,
+    "ondemand": false,
+    "configData":
+    {
+        "contentSource": "cti-api",
+        "compressionType": "raw",
+        "versionedContent": "cti-api",
+        "deleteDownloadedContent": true,
+        "url": "https://cti.wazuh.com/api/v1/catalog/contexts/test_context/consumers/test_consumer",
+        "outputFolder": "/tmp/output_folder",
+        "contentFileName": "content.json",
+        "databasePath": "/tmp/content_updater/rocksdb"
+    }
+}
+```
+
+> For simplicity, only the usage case-related configurations are shown.
+
+The configuration above will make the Content Manager launch each `10` seconds an orchestration that will download the content offsets from the Wazuh CTI API (context: `test_context`, consumer: `test_consumer`). The Content Manager will store the offsets, by groups of 1000, into output files located in `/tmp/output_folder`.
+
+Executing the Content Manager for the first time, with a starting offset of `975000`, will download from offset `975000` to the last available offset (`978576` in this example).
+
+```bash
+# ./content_manager_test_tool 
+ActionOrchestrator - Starting process
+API offset to be used: 975000
+Output folders created.
+FactoryContentUpdater - Starting process
+Creating 'cti-api' downloader
+Creating 'raw' content decompressor
+Creating 'cti-api' version updater
+Downloaded content cleaner created
+FactoryContentUpdater - Finishing process
+ActionOrchestrator - Finishing process
+ActionOrchestrator - Running process
+CtiApiDownloader - Starting
+CtiApiDownloader - Request processed successfully.
+CtiApiDownloader - Request processed successfully.
+CtiApiDownloader - Request processed successfully.
+CtiApiDownloader - Request processed successfully.
+CtiApiDownloader - Request processed successfully.
+CtiApiDownloader - Finishing
+SkipStep - Executing
+PubSubPublisher - Data published
+```
+
+The output files containing the downloaded offsets are available under `output_folder/contents/`, each of which contains 1000 offsets (the last one can contain fewer offsets if the total amount of offsets is not multiple of 1000).
```

Comment:
Question: The 1000 is hardcoded in the content manager? or is it a limitation of the API?

Reply:
Both. It is hardcoded because of an API limitation. Do you think it's worth to mention this?

The limitation is given by the Wazuh CTI API I believe. Maybe (we never know) a user wants to use its own CTI API, so a comment saying that "there is a limitation from the API" doesn't seem right to me.

Reply:
Thanks for the answer.
I think we can leave it as it is.